stellarphot/settings/models.py

```diff
@@ -54,6 +56,32 @@
 )
 
 
+# Make a type for a non-empty string for use in name fields
+# Names may not have any leading or trailing spaces, and cannot simply
+# be spaces. Though this could be implemented as a regular expression
+# pattern, that leads to a validation message likely to confuse users,
+# along the lines of (for pattern r"^\S$|^\S.*\S$"):
+#
+# String should match pattern "^\\S$|^\\S.*\\S$"
+#
+# Instead, the custom validation function below checks for errors and
+# raises a ValueError with a more user-friendly message.
+def _non_empty_string_validator(value):
+    if not value.strip():
+        raise ValueError("name must not be empty or contain only whitespace.")
+
+    if not re.search(r"^\S$|^\S.*\S$", value):
```

Comment:
So you are matching a SINGLE non-whitespace character as the string name or any string that starts and ends with non-whitespace characters?  I *guess* this makes sense, as I think of ways to avoid leading or trialing spaces... you have to allow for the possibility of a single character or multiple characters.  It does look odd, but the logic is sound.

Reply:
Yes, that is the intent. I thought about just stripping out whitespace silently but it seemed but to flag it instead.